client/src/lib/attempt-result.js

```diff
@@ -256,36 +256,82 @@ export function autocompleteTimeAttemptResult(time) {
   return roundOver10Mins(time);
 }
 
+/**
+ * Checks whether a given attempt is a world record of the given type.
+ * Returns the corresponding boolean.
+ */
+export function isWorldRecord(
+  attemptResult,
+  eventId,
+  type,
+  officialWorldRecords
+) {
+  const wr = officialWorldRecords?.find(
+    (wr) => wr.type === type && wr.event.id === eventId
+  )?.attemptResult;
+  return attemptResult <= wr;
```

Comment:
```suggestion
  const wr = officialWorldRecords.find(
    (wr) => wr.type === type && wr.event.id === eventId
  ) || null;
  return wr !== null && attemptResult <= wr.attemptResult;
```

Reply:
Suggested code was complaining when `officialWorldRecords === undefined` so I added a check in `isWorldRecord` that returns false when it is not defined.

Reply:
Existing tests caused `officialWorldRecords` to sometimes be undefined, so I added an empty array in those tests where the WR status of an attempt was not relevant to the test.